src/typing_extensions.py

```diff
@@ -659,7 +658,7 @@ def _proto_hook(cls, other):
                     isinstance(annotations, collections.abc.Mapping)
                     and attr in annotations
                     and issubclass(other, (typing.Generic, _ProtocolMeta))
-                    and other._is_protocol
+                    and getattr(other, "_is_protocol", False)
```

Comment:
Sorry for the breakage here. This was my mistake.

Subclasses of `typing.Generic` _are_ guaranteed to have an `_is_proto` class attribute on Python 3.8+, but not on Python 3.7. It was my fault for not spotting that when writing #184.

Could you possibly add an explanatory comment, something like this? It should help us maintain the code in the future

```suggestion
                    # All subclasses of Generic have an _is_proto attribute on 3.8+
                    # But not on 3.7
                    and getattr(other, "_is_protocol", False)
```

Reply:
Fixed